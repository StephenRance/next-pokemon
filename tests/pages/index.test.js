import Pokemon from '../../pages';
import { render, mount } from 'enzyme';

describe('<Pokemon />', () => {
	const props = {
		pokemon_results: [
			{
				name: 'Item 1'
			},
			{
				name: 'Item 2'
			},
			{
				name: 'Item 3'
			},
			{
				name: 'Item 4'
			}
		]
	};

	it('should match snapshot', () => {
		const target = render(<Pokemon {...props} />);

		expect(target).toMatchSnapshot();
	});

	it('should have 4 pokemon', () => {
		const target = mount(<Pokemon {...props} />);

		expect(target.find('Column')).toHaveLength(props.pokemon_results.length);
	});

	it('should have 0 pokemon', () => {
		props.pokemon_results = []
		const target = mount(<Pokemon {...props} />);

		expect(target.find('Column')).toHaveLength(0)
	});
});
