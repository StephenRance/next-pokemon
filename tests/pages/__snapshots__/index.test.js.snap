// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pokemon /> should match snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "Grid-sc-143fupq-0 choILo",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "Column-sc-3a2d7x-0 loLIxT",
          "width": "4",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "Card-xtwp70-0 dpXwBU",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "href": "/pokemon/Item 1",
                },
                "children": Array [
                  Object {
                    "data": "Item 1",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "href": undefined,
                },
                "x-attribsPrefix": Object {
                  "href": undefined,
                },
              },
            ],
            "name": "article",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "Column-sc-3a2d7x-0 loLIxT",
            "width": "4",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "Card-xtwp70-0 dpXwBU",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "href": "/pokemon/Item 2",
                  },
                  "children": Array [
                    Object {
                      "data": "Item 2",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "href": undefined,
                  },
                },
              ],
              "name": "article",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "Column-sc-3a2d7x-0 loLIxT",
              "width": "4",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "Card-xtwp70-0 dpXwBU",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "href": "/pokemon/Item 3",
                    },
                    "children": Array [
                      Object {
                        "data": "Item 3",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "a",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "href": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "href": undefined,
                    },
                  },
                ],
                "name": "article",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "Column-sc-3a2d7x-0 loLIxT",
                "width": "4",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "Card-xtwp70-0 dpXwBU",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "href": "/pokemon/Item 4",
                      },
                      "children": Array [
                        Object {
                          "data": "Item 4",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "a",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "href": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "href": undefined,
                      },
                    },
                  ],
                  "name": "article",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "width": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "width": undefined,
              },
            },
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "width": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "width": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "width": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "width": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "width": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "width": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "Column-sc-3a2d7x-0 loLIxT",
          "width": "4",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "Card-xtwp70-0 dpXwBU",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "href": "/pokemon/Item 2",
                },
                "children": Array [
                  Object {
                    "data": "Item 2",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "href": undefined,
                },
                "x-attribsPrefix": Object {
                  "href": undefined,
                },
              },
            ],
            "name": "article",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "Column-sc-3a2d7x-0 loLIxT",
            "width": "4",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "Card-xtwp70-0 dpXwBU",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "href": "/pokemon/Item 3",
                  },
                  "children": Array [
                    Object {
                      "data": "Item 3",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "href": undefined,
                  },
                },
              ],
              "name": "article",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "Column-sc-3a2d7x-0 loLIxT",
              "width": "4",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "Card-xtwp70-0 dpXwBU",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "href": "/pokemon/Item 4",
                    },
                    "children": Array [
                      Object {
                        "data": "Item 4",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "a",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "href": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "href": undefined,
                    },
                  },
                ],
                "name": "article",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "width": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "width": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "width": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "width": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "Column-sc-3a2d7x-0 loLIxT",
            "width": "4",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "Card-xtwp70-0 dpXwBU",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "href": "/pokemon/Item 1",
                  },
                  "children": Array [
                    Object {
                      "data": "Item 1",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "href": undefined,
                  },
                },
              ],
              "name": "article",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "width": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "width": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "width": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "width": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "Column-sc-3a2d7x-0 loLIxT",
          "width": "4",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "Card-xtwp70-0 dpXwBU",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "href": "/pokemon/Item 3",
                },
                "children": Array [
                  Object {
                    "data": "Item 3",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "href": undefined,
                },
                "x-attribsPrefix": Object {
                  "href": undefined,
                },
              },
            ],
            "name": "article",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "Column-sc-3a2d7x-0 loLIxT",
            "width": "4",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "Card-xtwp70-0 dpXwBU",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "href": "/pokemon/Item 4",
                  },
                  "children": Array [
                    Object {
                      "data": "Item 4",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "href": undefined,
                  },
                },
              ],
              "name": "article",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "width": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "width": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "Column-sc-3a2d7x-0 loLIxT",
            "width": "4",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "Card-xtwp70-0 dpXwBU",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "href": "/pokemon/Item 2",
                  },
                  "children": Array [
                    Object {
                      "data": "Item 2",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "href": undefined,
                  },
                },
              ],
              "name": "article",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "Column-sc-3a2d7x-0 loLIxT",
              "width": "4",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "Card-xtwp70-0 dpXwBU",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "href": "/pokemon/Item 1",
                    },
                    "children": Array [
                      Object {
                        "data": "Item 1",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "a",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "href": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "href": undefined,
                    },
                  },
                ],
                "name": "article",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "width": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "width": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "width": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "width": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "width": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "width": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "Column-sc-3a2d7x-0 loLIxT",
          "width": "4",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "Card-xtwp70-0 dpXwBU",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "href": "/pokemon/Item 4",
                },
                "children": Array [
                  Object {
                    "data": "Item 4",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "href": undefined,
                },
                "x-attribsPrefix": Object {
                  "href": undefined,
                },
              },
            ],
            "name": "article",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "Column-sc-3a2d7x-0 loLIxT",
            "width": "4",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "Card-xtwp70-0 dpXwBU",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "href": "/pokemon/Item 3",
                  },
                  "children": Array [
                    Object {
                      "data": "Item 3",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "href": undefined,
                  },
                },
              ],
              "name": "article",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "Column-sc-3a2d7x-0 loLIxT",
              "width": "4",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "Card-xtwp70-0 dpXwBU",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "href": "/pokemon/Item 2",
                    },
                    "children": Array [
                      Object {
                        "data": "Item 2",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "a",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "href": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "href": undefined,
                    },
                  },
                ],
                "name": "article",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "Column-sc-3a2d7x-0 loLIxT",
                "width": "4",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "Card-xtwp70-0 dpXwBU",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "href": "/pokemon/Item 1",
                      },
                      "children": Array [
                        Object {
                          "data": "Item 1",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "a",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "href": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "href": undefined,
                      },
                    },
                  ],
                  "name": "article",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "width": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "width": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "width": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "width": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "width": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "width": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "width": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "width": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
